src/internal/cache-client.ts

```diff
@@ -152,6 +154,47 @@ export class CacheClient {
     });
   }
 
+  public async setFetch(
+    cacheName: string,
+    setName: string
+  ): Promise<SetFetch.Response> {
+    validateCacheName(cacheName);
+    validateSetName(setName);
+    return await this.sendSetFetch(cacheName, this.convert(setName));
+  }
+
+  private async sendSetFetch(
+    cacheName: string,
+    setName: Uint8Array
+  ): Promise<SetFetch.Response> {
+    const request = new cache.cache_client._SetFetchRequest({
+      set_name: setName,
+    });
+    const metadata = this.createMetadata(cacheName);
+    return await new Promise(resolve => {
+      this.clientWrapper.getClient().SetFetch(
+        request,
+        metadata,
+        {
+          interceptors: this.interceptors,
+        },
+        (err, resp) => {
+          if (resp) {
+            if (resp.missing) {
+              resolve(new SetFetch.Missing());
+            } else if (resp.found) {
+              resolve(new SetFetch.Found(resp.found.elements));
+            } else {
+              resolve(new SetFetch.Error(cacheServiceErrorMapper(err)));
+            }
+          } else {
+            resolve(new SetFetch.Error(cacheServiceErrorMapper(err)));
+          }
+        }
```

Comment:
I ([meaning Stack Overflow](https://stackoverflow.com/a/75102198/14660)) found a better pattern.
```suggestion
        (err, resp) => {
          if (resp?.missing) {
            resolve(new CacheListFetch.Miss());
          } else if (resp?.found) {
            resolve(new CacheListFetch.Hit(resp.found.values));
          } else {
            resolve(new CacheListFetch.Error(cacheServiceErrorMapper(err)));
          }
        }
```